src/controllers/DeviceRegistrationController.ts

```diff
@@ -0,0 +1,104 @@
+import { Request, Response } from "express";
+import { sendJSONresponse } from "../common/Utils";
+import { Device } from "../models/DeviceModel";
+import * as winston from "winston";
+import { ISavedDevice } from "./Interfaces/IPusherController"
+
+export class DeviceRegistration {
+    register = (req: Request, res: Response) => {
+        const wallets = [...(new Set(req.body.wallets.map((wallet: string) => wallet.toLowerCase())))];
+        const inputPreferences = req.body.preferences || {};
+        const preferences = {
+            isAirdrop: inputPreferences.isAirdrop || false
+        }
+        const type: string = req.body.type || ""
+        const deviceID: string = req.body.deviceID
+        const token: string = req.body.token
+        const updateOptions = {
+            upsert: true,
+            new: true,
+            setDefaultsOnInsert: true
+        }
+        const updatesParams = {
+            deviceID,
+            wallets,
+            token,
+            preferences,
+            type
+        }
+
+        try {
+            if (deviceID && token) {
+                Device.findOneAndUpdate({deviceID}, updatesParams, updateOptions).then(registeredDevice => {
+                    this.sendOnRegisterSuccess(res, registeredDevice)
```

Comment:
the if and else blocks are duplicate code, you can minimize it.

Reply:
Done